src/Serializer.php

```diff
@@ -272,4 +277,24 @@ protected function getIncluded()
         $include = $this->request->get($this->expandParam);
         return is_string($include) ? preg_split('/\s*,\s*/', $include, -1, PREG_SPLIT_NO_EMPTY) : [];
     }
+
+    /**
+     * Checks Identification of resource object.
+     * http://jsonapi.org/format/#document-resource-object-identification
+     * @param ResourceInterface $resource
+     */
+    protected function checkIdentification(ResourceInterface $resource)
+    {
+        if (!is_string($resource->getId())) {
```

Comment:
@tuyakhov JSON-API clients **dont't** expect integers because of the specs specific requirement :-)

> Every resource object MUST contain an id member and a type member. The values of the id and type members MUST be strings.